src/librustc_lint/non_ascii_idents.rs

```diff
@@ -13,9 +15,100 @@ declare_lint! {
     "detects uncommon Unicode codepoints in identifiers"
 }
 
-declare_lint_pass!(NonAsciiIdents => [NON_ASCII_IDENTS, UNCOMMON_CODEPOINTS]);
+declare_lint! {
+    pub CONFUSABLE_IDENTS,
+    Allow,
```

Comment:
Oh I missed that you changed this, we need this at least at warn in order for the perf runs to be of any use -- want to switch it back and I can rerun perf?

Reply:
Sure, done, and rebased. 

At least from this perf run it seems that collecting identifiers at the parsing stage's performance impact is negligible.

Reply:
Wait, I thought lints always run? Or did we implement that optimization to turn off lints that are Allow where not needed?

Reply:
@Manishearth 
>Or did we implement that optimization to turn off lints that are Allow where not needed?

We did!

Reply:
We didn't but this one is special cased (https://github.com/rust-lang/rust/pull/71542/files#diff-ef2570d66772bb8e6caa51779efe25f2R107-R109) -- I believe that code may mean that it's only togglable at the crate root though, not sure.

Reply:
hmmm..

really wish lints could declare what their smallest unit of action is and have their passes just not be run based on that. all zero-sized lints would be free when `Allow`, then.